aspnetcore/security/authentication/certauth.md

```diff
@@ -206,3 +206,243 @@ Complete the following steps In IIS Manager:
 ### Azure and custom web proxies
 
 See the [host and deploy documentation](xref:host-and-deploy/proxy-load-balancer#certificate-forwarding) for how to configure the certificate forwarding middleware.
+
+### Use certificate authentication with the X-ARR-ClientCert header
+
+The `AddCertificateForwarding` method is used to specify:
+
+* The client header name.
+* How the certificate is to be loaded (using the `HeaderConverter` property).
+
+When sending the certificate with the `HttpClient` using the default settings, the `ClientCertificate` will always be null. The `X-ARR-ClientCert` header is used to pass the client certificate. The certificate is passed as a string to work around this problem.
+
+```csharp
+services.AddCertificateForwarding(options =>
+{
+	options.CertificateHeader = "X-ARR-ClientCert";
+	options.HeaderConverter = (headerValue) =>
+	{
+		X509Certificate2 clientCertificate = null;
+		if(!string.IsNullOrWhiteSpace(headerValue))
+		{
+			byte[] bytes = StringToByteArray(headerValue);
+			clientCertificate = new X509Certificate2(bytes);
+		}
+
+		return clientCertificate;
+	};
+});
+```
+
+The `Startup.Configure` method then adds the middleware. `UseCertificateForwarding` is called before the calls to `UseAuthentication` and `UseAuthorization`.
+
+```csharp
+public void Configure(IApplicationBuilder app, IWebHostEnvironment env)
+{
+	...
+	
+	app.UseRouting();
+
+	app.UseCertificateForwarding();
+	app.UseAuthentication();
+	app.UseAuthorization();
+
+	app.UseEndpoints(endpoints =>
+	{
+		endpoints.MapControllers();
+	});
+}
+```
+
+A separate class can be used to implement validation logic. Because the same self-signed certificate is used in this example, ensure that only your certificate can be used. Validate that the thumbprints of both the client certificate and the server certificate match, otherwise any certificate can be used and will be enough to authenticate. This would be used inside the `AddCertificate` method. You could also validate the subject or the issuer here if you're using intermediate or child certificates.
+
+```csharp
+using System.IO;
+using System.Security.Cryptography.X509Certificates;
+
+namespace AspNetCoreCertificateAuthApi
+{
+    public class MyCertificateValidationService
+    {
+        public bool ValidateCertificate(X509Certificate2 clientCertificate)
+        {
+            var cert = new X509Certificate2(Path.Combine("sts_dev_cert.pfx"), "1234");
```

Comment:
No hard coded passwords. I know it's an example, so just hardcode a thumbprint :)

Reply:
Added text:
Do not hardcode passwords in production code, use thumbprint or key vault

If you want, I could also change this to use the local store to get the thumbprint